include/splinepy/splines/helpers/scalar_type_wrapper.hpp

```diff
@@ -200,6 +200,14 @@ bool ScalarTypeInsertKnot(SplineType& spline,
   // see if it was successful
   const auto& knot_vector =
       *spline.GetParameterSpace().GetKnotVectors()[query_dim];
+
+  // Check if request is valid before SplineLib
+  const auto para_bounds = GetParametricBounds(spline);
+  if ((para_bounds[query_dim][0] >= query)
+      || (para_bounds[query_dim][0] <= query)) {
+    return false;
+  }
+
```

Comment:
@j042 This was added in an attempt to get a quick fix. But apparently, c^{-1} knot insertion is not working. Do you have any insights as to why? I think that demands your expertise

Reply:
I did a stupid copy pasting mistake. I think it's fine. Still : Please review this part of the code and tell me if you rather want checks on the `SplineLib`-side